src/java.base/share/classes/sun/nio/ch/ChannelOutputStream.java

```diff
@@ -64,10 +64,15 @@ WritableByteChannel channel() {
      * If the channel is selectable then it must be configured blocking.
      */
     private void writeFully(ByteBuffer bb) throws IOException {
-        while (bb.remaining() > 0) {
+        int pos = bb.position();
+        int rem = bb.limit() - pos;
+        while (rem > 0) {
+            bb.limit(pos + Math.min(Streams.MAX_BUFFER_SIZE, rem));
             int n = ch.write(bb);
             if (n <= 0)
-                throw new RuntimeException("no bytes written");
+                throw new IOException("Write failed");
```

Comment:
It would be a broken WritableByteChannel implementation to return < 0. If it were a non-blocking then IllegalBlockingModeException would be thrown already. So I think this throw IOException is okay here but it is a sign of a broken channel implementation.